pkg/providers/version/version.go

```diff
@@ -129,3 +121,25 @@ func validateK8sVersion(v string) error {
 
 	return nil
 }
+
+func (p *DefaultProvider) getEKSVersion(ctx context.Context) (string, error) {
+	output, err := p.eksapi.DescribeCluster(ctx, &eks.DescribeClusterInput{
+		Name: lo.ToPtr(options.FromContext(ctx).ClusterName),
+	})
+	if err != nil {
+		return "", err
+	}
+	version := lo.FromPtr(output.Cluster.Version)
+	if version == "" {
+		return "", fmt.Errorf("failed to get cluster version")
+	}
+	return version, nil
```

Comment:
Discussed a bit offline you and @jonathan-innis , we can assume if we don't get an error from the API that the version is valid. 
```suggestion
	return lo.FromPtr(output.Cluster.Version), nil
```